services/ui_backend_service/api/utils.py

```diff
@@ -213,6 +215,130 @@ def builtin_conditions_query_dict(query: MultiDict):
 }
 
 
+def bound_filter(op, term, key):
+    "returns function that binds the key, and the term that should be compared to, on an item"
+    _filter = array_filter_ops[op]
+    return lambda item: _filter(item[key], term) if key in item else False
+
+
+# NOTE: keep these as simple comparisons,
+# any kind of value decoding should be done outside the lambdas instead
+# to promote reusability.
+array_filter_ops = {
+    "eq": (lambda item, term: str(item) == term),
+    "ne": (lambda item, term: str(item) != term),
+    "lt": (lambda item, term: int(item) < int(term)),
```

Comment:
Any concerns when comparing string values here? Thing that comes to mind first would be run_id/run_number and task_id/task_number

Reply:
And what happens if we want to cast non-numerical values to integer, are we catching errors in safe manner,

Reply:
good point on the type casting gotcha. the 'term' in the filters will originate from url params, so this will be a string by default to my knowledge, so comparing values as strings by default should not be a major issue here. I'll add a catch for the possible type-cast failures